File: hs/21.hs
-- 31626
import Euler(divisorPowerSum)

nn = 10000

-- divisorPowerSum 1 is the sum of all divisors
isAmicable a = a /= b && a == factorSum b
    where b = factorSum a
          factorSum n = divisorPowerSum 1 n - n
sumAmicable n = sum $ filter isAmicable [2..n]

main = putStrLn $ show $ sumAmicable nn

File: py/21.py
# 31626
import euler
N = 10000
s = 0
for a in xrange(2,N):
    b = sum(euler.all_factors(a)[:-1])
    db = sum(euler.all_factors(b)[:-1])
    if a != b and a == db:
        s += a
print s
